hummingbot/connector/exchange/ndax/ndax_exchange.py

```diff
@@ -65,24 +76,193 @@ def __init__(self,
+
+    async def _update_balances(self):
+        """
+        Calls REST API to update total and available balances
+        """
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+
+        params = {
+            "OMSId": 1,
+            "AccountId": self.account_id
+        }
+        account_positions: List[Dict[str, Any]] = await self._api_request(
+            method="GET",
+            path_url=CONSTANTS.ACCOUNT_POSITION_PATH_URL,
+            params=params,
+            is_auth_required=True
+        )
+        for position in account_positions:
+            asset_name = position["ProductSymbol"]
+            self._account_balances[asset_name] = Decimal(str(position["Amount"]))
+            self._account_available_balances[asset_name] = self._account_balances[asset_name] - Decimal(str(position["Hold"]))
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+    async def _update_order_status(self):
+        # Waiting on buy and sell functionality.
+        pass
+
+    async def _status_polling_loop(self):
+        """
+        Periodically update user balances and order status via REST API. This serves as a fallback measure for web
+        socket API updates.
+        """
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+                await safe_gather(
+                    self._update_balances(),
+                    self._update_order_status(),
+                )
+                self._last_poll_timestamp = self.current_timestamp
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
+                self.logger().error(str(e), exc_info=True)
+                self.logger().network("Unexpected error while fetching account updates.",
+                                      exc_info=True,
+                                      app_warning_msg="Could not fetch account updates from NDAX. "
+                                                      "Check API key and network connection.")
+                await asyncio.sleep(0.5)
+
+    def tick(self, timestamp: float):
+        """
+        Is called automatically by the clock for each clock tick(1 second by default).
+        It checks if a status polling task is due for execution.
+        """
+        now = time.time()
+        poll_interval = (self.SHORT_POLL_INTERVAL
+                         if now - self._user_stream_tracker.last_recv_time > 60.0
+                         else self.LONG_POLL_INTERVAL)
+        last_tick = int(self._last_timestamp / poll_interval)
+        current_tick = int(timestamp / poll_interval)
+        if current_tick > last_tick:
```

Comment:
Maybe we could also add to the update poll execution the elapsed time since the last message received through the websocket (to avoid doing unnecessary request and wasting the API limits

Reply:
@aarmoa So I have been thinking about this. I think there is still a need to have trigger the status polling task at regular intervals and it shouldnt be dependant on the elapsed time of the last message received by the websocket. 

One of the scenarios where the regular interval polling task is necessary is when for some reason the websocket is only not receiving balance updates, but is still receiving other user account events. The bot still requires a fallback to ensure that it has the updated user balances.